### PR TITLE
Support for PHP 7.2

### PR DESCRIPTION
**Pull request by escopecz**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Already covered
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5769 (closed)
| BC breaks? | N
| Deprecations? | Y

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
It's about time to prepare Mautic for PHP 7.2. This PR upgrades the maximum PHP contstants and fixes PHP warning about Mcrypt constants.

# WARNING
**If you have some plugin credentials stored by Mcrypt they cannot be decrypted by OpenSSL. Make sure you have both libraries installed and enabled. Then re-save all plugins so the credentials are encrypted by OpenSSL and then you can upgrade to PHP 7.2.**

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Upgrade your PHP to 7.2.
2. You'll see a message that this version is not supported.
3. Hack `MAUTIC_MAXIMUM_PHP` in VersionCheckMiddleware.php to `7.2.999` to overcome the message.
4. Clear the cache.
5. In dev mode you'll see warning everywhere.

#### Steps to test this PR:
1. Apply this PR
2. No warnings are visible. Everything looks normal and little faster :)

#### List deprecations along with the new alternative:
1. Mcrypt extension does not exist on PHP 7.2. Use OpenSSL instead.

### CODE REVIEW

**Comment by Gregy** [comment]
What about creating a command that would go through all the encrypted fields and resave them in openssl format? That would make upgrading easier and it would be possible to automate it.

File: app/middlewares/VersionCheckMiddleware.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ class VersionCheckMiddleware implements HttpKernelInterface, PrioritizedMiddlewa
     const PRIORITY = 10;
 
     const MAUTIC_MINIMUM_PHP = '5.6.19';
-    const MAUTIC_MAXIMUM_PHP = '7.1.999';
+    const MAUTIC_MAXIMUM_PHP = '7.2.999';
```
[the remark this answers is not in the document]
Comment (escopecz) [reply app/middlewares/VersionCheckMiddleware.php]: Added.

[the remark this answers is not in the document]
Comment (helbertqueiroz) [reply app/middlewares/VersionCheckMiddleware.php]: Thanks!!

**Comment by andersonbraz** [comment]
I solved like this:

On file: mautic/app/middlewares/VersionCheckMiddleware.php

I changing:

``php
const MAUTIC_MAXIMUM_PHP = '7.1.999';
``

to

``php
const MAUTIC_MAXIMUM_PHP = '7.2.999';
``

Best Regards!

by [Anderson Braz](https://andersonbraz.com)

**Comment by escopecz** [comment]
@andersonbraz Be careful. There are more changes required for PHP 7.2 as you can see in this PR and more needs to be made.

File: app/middlewares/VersionCheckMiddleware.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (javiergas) [reply app/middlewares/VersionCheckMiddleware.php]: I think there also might be a problem with this:

mautic.WARNING: PHP Warning - count(): Parameter must be an array or an object that implements Countable - in file /var/www/contact/app/bundles/CoreBundle/Model/IteratorExportDataModel.php - at line 71

**Comment by RCheesley** [comment]
Tested this PR and I don't see any further errors in the log once the PR is applied.

**Comment by StudioMaX** [comment]
When this update will be released it will be necessary to add support for the upcoming PHP 7.3 😃

**Comment by escopecz** [comment]
@galvani this PR is far from ready. Notice the failed tests and that's just known issues.

**Comment by escopecz** [comment]
Closing in favour of https://github.com/mautic/mautic/pull/6599

**Pull request closed without merging**